app/pods/search/newsletter-infos/route.js

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable class-methods-use-this */
+import Route from '@ember/routing/route';
+import { isEmpty, isPresent } from '@ember/utils';
+import moment from 'moment';
+import search from 'frontend-kaleidos/utils/mu-search';
+import Snapshot from 'frontend-kaleidos/utils/snapshot';
+import { inject as service } from '@ember/service';
+
+export default class AgendaitemSearchRoute extends Route {
+  @service metrics;
+  queryParams = {
+    page: {
+      refreshModel: true,
+      as: 'pagina',
+    },
+    size: {
+      refreshModel: true,
+      as: 'aantal',
+    },
+    sort: {
+      refreshModel: true,
+      as: 'sorteer',
+    },
+  };
+
+  textSearchFields = Object.freeze(['title', 'subTitle', 'richtext']);
+
+  constructor() {
+    super(...arguments);
+    this.lastParams = new Snapshot();
+  }
+
+  postProcessAgendaitems(newsletter) {
+    const agendaitems = newsletter.attributes.agendaitems;
+    if (agendaitems) {
+      if (Array.isArray(agendaitems)) {
+        for (const agendaitem of agendaitems) {
+          if (agendaitem['nextVersionId'] == null) {
+            // there is no next version = latest agendaitem
+            return (newsletter.attributes.latestAgendaitem = agendaitem);
```

Comment:
Is het returnen van een assignment iets dat door prettier of dergelijke aangeraden wordt? Ik herken het van m'n C-tijd, maar 't is me geen beken patroon in JS?

Reply:
Die return is er gewoon om uit de lus te springen, kan op zich ook na de assignment